Phone field documentation (#388)
Laravel-Backpack/docs: 5.x/crud-fields.md

> NOTE: if you are having trouble uploading big images, please check your php extensions **apcu** and/or **opcache**, users have reported some issues with these extensions when trying to upload very big images. REFS: https://github.com/Laravel-Backpack/CRUD/issues/3457

<hr>

<a name="phone"></a>
### phone <span class="badge badge-pill badge-info">PRO</span>

> Show a telephone input and let the user choose a flag from dropdown.

Reviewer: ```suggestion
Show a telephone number input. Lets the user choose the prefix using a flag from dropdown.
```

```php
[   // phone
    'name'  => 'phone', // db column for phone
    'label' => 'Phone',
    'type'  => 'phone',

    // OPTIONALS

>     'initialCountry' => 'cl',
>     'separateDialCode' => true,
>     'nationalMode' => true,
>     'autoHideDialCode' => false,
>     'placeholderNumberType' => 'MOBILE',
>     // most options provided by intlTelInput.js are supported, you can try them out using the `config` attribute;
>     //  take note that options defined in `config` will override any default values from the field;
>     ```
>     'config' => [
>         'onlyCountries' => ['pt', 'ro'],
>     ]
> ],

Reviewer: ```suggestion
    // most options provided by intlTelInput.js are supported, you can try them out using the `config` attribute;
    //  take note that options defined in `config` will override any default values from the field;
    'config' => [
        'onlyCountries' => ['pt', 'ro'],
        'initialCountry' => 'cl',
        'separateDialCode' => true,
        'nationalMode' => true,
        'autoHideDialCode' => false,
        'placeholderNumberType' => 'MOBILE',
    ]
],
```

>     // inside the config attribute; take note that the directly defined attributes will override anything defined in the config;

Reviewer: I've just suggested changing the merge order in https://github.com/Laravel-Backpack/PRO/pull/103/files so this could become:

```suggestion
    // inside the config attribute; take note that attributes defined in `config` will override any defaults or previously set values;
    ```

```

> Please note it is recommended that you check [official documentation](https://github.com/jackocnr/intl-tel-input) to get parameters description.

Reviewer: ```suggestion
For more info about parameters please see this JS plugin's [official documentation](https://github.com/jackocnr/intl-tel-input).
```

> NOTE: if you are having trouble uploading big images, please check your php extensions **apcu** and/or **opcache**, users have reported some issues with these extensions when trying to upload very big images. REFS: https://github.com/Laravel-Backpack/CRUD/issues/3457

<hr>

<a name="phone"></a>
### phone <span class="badge badge-pill badge-info">PRO</span>

Show a telephone number input. Lets the user choose the prefix using a flag from dropdown.

```php
[   // phone
    'name'  => 'phone', // db column for phone
    'label' => 'Phone',
    'type'  => 'phone',

>     // OPTIONALS

Reviewer 2: ```suggestion
    //  DEFAULTS
```

    'initialCountry' => 'cl',
    'separateDialCode' => true,
    'nationalMode' => true,
    'autoHideDialCode' => false,
    'placeholderNumberType' => 'MOBILE',

>     // the above are some officially supported and tested options, provided by intlTelInput.js;
>     // there are more, we just haven't tested them; check out their docs - you can use any of them 
>     // inside the config attribute; take note that attributes defined in `config` will override any defaults or previously set values;

Reviewer 2: ```suggestion
    // most options provided by intlTelInput.js are supported, you can try them out using the `config` attribute;
    //  take note that options defined in `config` will override any default values from the field;
```

> NOTE: if you are having trouble uploading big images, please check your php extensions **apcu** and/or **opcache**, users have reported some issues with these extensions when trying to upload very big images. REFS: https://github.com/Laravel-Backpack/CRUD/issues/3457

<hr>

<a name="phone"></a>
### phone <span class="badge badge-pill badge-info">PRO</span>

Show a telephone input and let the user choose a flag from dropdown.

```php
[   // phone
    'name'  => 'phone', // db column for phone
    'label' => 'Phone',
    'type'  => 'phone',

    // OPTIONALS
    // options offically supported and tested, this options will be prioritized to config parameters
    // this mean if you send parameter "initialCountry" => "us" 

>     // and "config" => ["initialCountry" => "us"], "cl" value will be used

Reviewer: I'd move this note AFTER the "officially supported" options. They only need to know about that if they wanna go deeper, so that's a better moment to say it I think, and we introduce them to `config` then.

So I'd rephrase to:

_the above are some officially supported and tested options, provided by intlTelInput.js; there are more, we just haven't tested them; check out their docs - you can use any of them inside the config attribute; take note that the directly defined attributes will override anything defined in the config;_

    'initialCountry' => 'cl',
    'separateDialCode' => true,
    'nationalMode' => true,
    'autoHideDialCode' => false,
    'placeholderNumberType' => 'MOBILE',
    // options sent to intlTelInput.min.js, check documentation for details
    'config' => [
        'onlyCountries' => ['pt', 'ro'],
    ]
],
```

Please note it is recommended that you check [official documentation](https://github.com/jackocnr/intl-tel-input) to get parameters description.

Your end result will look like this:

> Input preview:

Reviewer: These two lines say the same thing, so let's just choose one of them.

![CRUD Field - phone](https://user-images.githubusercontent.com/1032474/204588174-48935030-54e6-4a30-b34c-7e94220ae242.png)

> 

Reviewer: I think it'd be good to also introduce people to the Phone Validation package here. Just like a sidenote, something like "you can validate this using Laravel's default `numeric` or if you want something advanced, we recommend _link_"
